core/src/main/java/io/grpc/Metadata.java

```diff
@@ -89,6 +91,27 @@ public String parseAsciiString(String serialized) {
   };
 
   /**
+   * Simple metadata marshaller that encodes strings as is.
+   *
+   * <p>This should be used with ASCII strings that only contain the characters listed in the class
+   * comment of {@link AsciiMarshaller}. Otherwise the output may be considered invalid and
+   * discarded by the transport, or the call may fail.
+   */
+  public static final AsciiMarshaller<List<String>> ASCII_LIST_MARSHALLER =
+      new AsciiMarshaller<List<String>>() {
+
+    @Override
+    public String toAsciiString(List<String> values) {
+      return Joiner.on(',').join(values);
```

Comment:
Well, comma is required by the spec.  it also leaves no room for a space, which http2 does support I think.  Im personally okay with taking multiple header values as in, getting a list back out.  Also possible is splitting out csv's on parsing metadata.